### PR TITLE
Fix linter issues in smtpd package

### PR DESCRIPTION
**Pull request by hairyhenderson**
Followup from #58 - a pile of lint fixes.

I cleared each one in a single commit to make this _slightly_ easier to review. The last one (for `govet`) is a doozy though, apologies to the reviewer 😅

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -104,7 +104,7 @@ issues:
         # - gocritic
         # - gofmt
         # - goimports
-        - gosec
+        # - gosec
```
Comment (owensmallwood) [inline .golangci.yml]: Is this one really worth it if we're pretty much ignoring it everywhere anyways?

Comment (hairyhenderson) [reply .golangci.yml]: IMO yes - it can catch some pretty serious bugs

File: internal/smtpd/protocol.go [hunk 2]
```diff
@@ -23,10 +23,11 @@ type command struct {
 	params []string
 }
 
-func parseLine(line string) (cmd command) {
```
Comment (owensmallwood) [inline internal/smtpd/protocol.go:25]: Good riddance! I don't like these

File: internal/smtpd/example_test.go [hunk 3]
```diff
@@ -1,44 +0,0 @@
...
-
-import (
-	"errors"
-	"net/smtp"
-	"strings"
-
-	"github.com/grafana/smtprelay/internal/smtpd"
-)
-
-func ExampleServer() {
```
Comment (owensmallwood) [inline internal/smtpd/example_test.go:11]: What was the point of this anyways?

**Review by owensmallwood — APPROVED** [review]
LGTM 👍

**Pull request merged**